github/PaginatedList.py

```diff
@@ -134,8 +134,20 @@ def __init__(self, contentClass, requester, firstUrl, firstParams, headers=None,
     @property
     def totalCount(self):
         if not self.__totalCount:
-            self._grow()
-
+            params = {} if self.__nextParams is None else self.__nextParams.copy()
+            params.update({"per_page": 1, "page": 1})
+            headers, data = self.__requester.requestJsonAndCheck(
+                "GET",
+                self.__firstUrl,
+                parameters=params,
+                headers=self.__headers
+            )
+            if 'link' not in headers:
+                self.__totalCount = len(data)
+            else:
+                links = self.__parseLinkHeader(headers)
+                lastUrl = links.get("last")
+                self.__totalCount = int(re.match(r'.*[&\?]page=(\d*).*?', lastUrl).group(1))
```

Comment:
Isn't it better to use `urlparse` to find the `query` part of the lastURL and then using `parse_qs(l)` to find the value of `page`, than using your own regex?

Reply:
I agree, this is better